src/widgets/ot_carousel.eliom

```diff
@@ -163,6 +197,26 @@ let%shared make
     let pos_set = ~%pos_set in
     let action = ref (`Move (0, 0)) in
     let animation_frame_requested = ref false in
+
+    (* fill pages with content when switching to for the first time *)
+    let () = match ~%contents with | Strict _ -> () | Lazy gen_contents ->
+      let mk_generator i gen = ref @@ if i = ~%position then None else Some gen in
+      (* None: no generator as content is already present *)
+      let content_generators = List.mapi mk_generator gen_contents in
+      let thread = pos_signal |> React.S.map @@ fun i -> Lwt.async @@ fun () ->
+        let content_generator = List.nth content_generators i in
+        match !content_generator with
+          | Some gen ->
+              content_generator := None;
+              let%lwt new_content = gen () in
+              let old_content = List.nth ~%initial_contents i in
```

Comment:
`~%initial_contents` is going to be costly if some of its elements are not building using the `Eliom_content.D` module.